smoke-test/src/main/java/org/opennms/smoketest/containers/OpenNMSContainer.java

```diff
@@ -467,6 +470,11 @@ public int getGeneratedUserId() {
 
     @Override
     public void afterTest(final TestDescription description, final Optional<Throwable> throwable) {
+        if (afterTestCalled) {
```

Comment:
This is odd; should we be overriding something else instead? How does this happen?

Reply:
> This is odd

I can definitely confirm that this is odd, and it kinda drove me nuts. :)

> should we be overriding something else instead? How does this happen?

I spent a moderate amount of time digging in, and I couldn't figure out why it was happening, so I punted with this workaround. I was able to figure out that it appears the Rule that leads to this is getting called twice by JUnit: once before the container is torn down, once after, at which point we get failures, even with a successful execution.

Except, I just double-checked and this doesn't appear to be happening on `develop` and `release-31.x`. Hrm. I wonder if something else I tweaked caused this or something with the testcontainers update (or maybe it only happens on failures). If you're okay with merging this now, I can continue digging in on this on the side.

Reply:
Except now, when I test, it doesn't happen at all. Hrrrrmm.... I'll call this one okay for now.